docs/user-guide/deploying-application/image-labels-and-comments.md

```diff
@@ -0,0 +1,115 @@
+* Add `hotfix image only` label to indicate a one-time patch on production.
+* Comments like `This image is buggy and shouldn't be used for deployment` to caution other users from deploying an unwanted image.
+
+![Figure 1: Labels and Comments](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-and-comment.jpg)
+
+Such labels and comments will be visible only within Devtron, and will not propagate to your [container registry](../../reference/glossary.md#containeroci-registry) (say Docker Hub), unlike custom [image tag pattern](../creating-application/workflow/ci-pipeline.md#custom-image-tag-pattern). You may use it to simplify the management and [selection of container images](./triggering-cd.md#deploying-approved-image) for deployment.
+
+{% hint style="warning" %}
+Tagging labels and comments are supported only for images in workflows with at least one production deployment pipeline. In Devtron, you can go to **Global Configurations** → **Clusters & Environments** to identify a production environment by checking the 'Prod' label.
+{% endhint %}
+
+---
+
+## Adding Labels & Comments
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have [Build & deploy permission](../global-configurations/authorization/user-access.md#role-based-access-levels) or above (along with access to the environment and application) to add labels and comments.
+{% endhint %}
+
+You can add labels and comments from the following pages:
+
+* [From Build & Deploy](#from-build--deploy)
+* [From Build History](#from-build-history)
+* [From Deployment History](#from-deployment-history) (only after deployment)
+* [From App Details](#from-app-details) (only after deployment)
+
+{% hint style="warning" %}
+You can add multiple labels to an image. Once added, they cannot be used for another image. <br />
+Refer [Deleting Labels](#deleting-labels--comments) if you commit a mistake while adding labels.
+{% endhint %}
+ 
+### From Build & Deploy
+
+![Figure 2: Adding Labels and Comments - 'Build & Deploy' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-build.gif)
+
+### From Build History
+
+![Figure 3: Adding Labels and Comments - 'Build History' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-build-history.gif)
+
+### From Deployment History
+
+![Figure 4: Adding Labels and Comments - 'Deployment History' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-deployment-history.gif)
+
+### From App Details
+
+![Figure 5: Adding Labels and Comments - 'App Details' Page](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/deploying-application/tag-comment/tag-app-details.gif)
+
+---
+
+## Deleting Labels & Comments
+
+### Soft-Delete Labels
```

Comment:
We should also mention that soft deleted labels cannot be reused. Only hard deleted labels can be reused. Please verify this.

Reply:
Verified. Added as well. Thanks!